service/history/shard/controller.go

```diff
@@ -255,6 +260,15 @@ func (c *controller) shardClosedCallback(shardID int, shardItem *historyShardsIt
 }
 
 func (c *controller) getOrCreateHistoryShardItem(shardID int) (*historyShardsItem, error) {
+	if shardID >= c.config.NumberOfShards { // zero based shard ID
```

Comment:
nit: will shardID < 0?

Reply:
good call out. We should check it as well